src/containers/User/userInfo.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { connect } from "react-redux";
+import { push, replace } from "connected-react-router";
+import { routes } from "../Router";
+import Header from "../../components/Header/Header";
+import Footer from '../../components/Footer/Footer';
+import {
+  WrapperProfile,
+  CustomDiv,
+  DarkDiv,
+  TextProfile,
+} from "../../components/globalStyle";
+import { getUserInfo, getOrdersHistory } from "../../actions/profile";
+import editIcon from "../../img/edit.svg";
+import OrderCard from '../../components/OrderCard';
+
+class Profile extends React.Component {
+  state = {};
+
+  componentDidMount() {
+    const token = localStorage.getItem("token");
+    this.props.getUserInfo(token);
+    this.props.getOrdersHistory();
+    
+    if(token === null){
+      this.props.goToLogin()
+    }
+  }
+
+  render() {
+    const { user } = this.props;
+    const ordersToString = JSON.stringify(this.props.orders);
+
+    return (
+      <WrapperProfile>
+        <Header title={"Meu perfil"} goBack={false} />
+
+        <CustomDiv>
+          <p>
+            {user && user.name} <img src={editIcon} onClick={() => this.props.goToUpdateUserInfo()} />
+          </p>
+          <p>{user && user.email}</p>
+          <p>{user && user.cpf}</p>
+        </CustomDiv>
+
+        <DarkDiv>
+          <span>Endereço cadastrado</span>
+          <p>
+            {user && user.address} <img src={editIcon} onClick={() => this.props.goToUpdateAddress()}/>
+          </p>
+        </DarkDiv>
+
+        <TextProfile>Histórico de Pedidos</TextProfile>
+
+        <hr />
+
+        {ordersToString === "[]" || ordersToString === undefined ? (
```

Comment:
seria mais rápido checar assim aqui: `this.props.orders && this.props.orders.length === 0`